host-controller/src/main/java/org/jboss/as/domain/controller/operations/SyncServerStateOperationHandler.java

```diff
@@ -126,30 +148,51 @@ public void handleResult(OperationContext.ResultAction resultAction, OperationCo
         }, OperationContext.Stage.MODEL);
     }
 
-    private Map<String, Boolean> determineServerStateChanges(OperationContext context, Resource domainRootResource, ServerOperationResolver resolver,
-                                                             ContentDownloader contentDownloader) {
-        final Map<String, Boolean> serverStateChanges = new HashMap<>();
+    private Map<String, SyncServerResultAction> determineServerStateChanges(final OperationContext context, final Resource domainRootResource, final ServerOperationResolver resolver,
+                                                                            final ContentDownloader contentDownloader) {
+        final Map<String, SyncServerResultAction> serverStateChanges = new HashMap<>();
         for (ModelNode operation : operations) {
-            PathAddress addr = PathAddress.pathAddress(operation.get(OP_ADDR));
+            final PathAddress addr = PathAddress.pathAddress(operation.get(OP_ADDR));
             contentDownloader.checkContent(operation, addr);
-
-            Map<Set<ServerIdentity>, ModelNode> serverMap = resolver.getServerOperations(context, operation, addr);
+            final ImmutableManagementResourceRegistration registration = context.getResourceRegistration();
+            final Map<Set<ServerIdentity>, ModelNode> serverMap = resolver.getServerOperations(context, operation, addr);
             for (Map.Entry<Set<ServerIdentity>, ModelNode> entry : serverMap.entrySet()) {
-                ModelNode op = entry.getValue();
-                boolean restart = op.get(OP).asString().equals(ServerProcessStateHandler.REQUIRE_RESTART_OPERATION);
-                for (ServerIdentity id : entry.getKey()) {
-                    String serverName = id.getServerName();
-                    Boolean existing = serverStateChanges.get(serverName);
-                    if (existing == null || (!existing && restart)) {
-                        serverStateChanges.put(serverName, restart);
+                final ModelNode op = entry.getValue();
+                final String opName = op.get(OP).asString();
```

Comment:
@luck3y I think 2) is fine if 1) is uncomfortable at all. For any op in profile=*/subystem=*, ServerOperationResolver is *not* going to change the operation name, so the 2) type check isn't going to affect anything for those ops. For other resources, the odds of ServerOperationResolver  changing the name but not returning REQUIRE_RESTART_OPERATION or REQUIRE_RELOAD_OPERATION are really low, and if it happens we'd just require restart which is hardly the end of the world if reload was sufficient.